openedx/core/djangoapps/user_authn/views/login_form.py

```diff
@@ -186,16 +187,26 @@ def login_and_registration_form(request, initial_mode="login"):
         except (KeyError, ValueError, IndexError) as ex:
             log.exception("Unknown tpa_hint provider: %s", ex)
 
-    # Redirect to authn MFE if it is enabled or user is not an enterprise user or not coming from a SAML IDP.
+    # Redirect to authn MFE if it is enabled
+    # AND
+    #   user is not an enterprise user
+    # AND
+    #   tpa_hint_provider is not available
+    # AND
+    #   user is not coming from a SAML IDP.
     saml_provider = False
     running_pipeline = pipeline.get(request)
-    enterprise_customer = enterprise_customer_for_request(request)
     if running_pipeline:
         saml_provider, __ = third_party_auth.utils.is_saml_provider(
             running_pipeline.get('backend'), running_pipeline.get('kwargs')
         )
 
-    if should_redirect_to_authn_microfrontend() and not enterprise_customer and not saml_provider:
+    enterprise_customer = enterprise_customer_for_request(request)
+
+    if should_redirect_to_authn_microfrontend() and \
+            not enterprise_customer and \
+            not tpa_hint_provider and \
```

Comment:
Are there any unit tests that should be added/updated to account for this change in logic?

Reply:
yes good call, I did check that this unit test still covers this code logic for the case of tpa_hint_provider not set

```
root@lms:/edx/app/edxapp/edx-platform# pytest openedx/core/djangoapps/user_authn/views/tests/test_logistration.py::LoginAndRegistrationTest
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.8.5, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
django: settings: lms.envs.test (from ini)
Using --randomly-seed=605493110
rootdir: /edx/app/edxapp/edx-platform, configfile: setup.cfg
plugins: metadata-1.8.0, json-report-1.2.4, cov-2.11.1, xdist-2.2.1, attrib-0.1.3, django-4.1.0, forked-1.3.0, coverage-pytest-plugin-0.1, celery-4.4.7, randomly-3.6.0, Faker-8.0.0
collected 47 items

openedx/core/djangoapps/user_authn/views/tests/test_logistration.py
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /edx/app/edxapp/edx-platform/openedx/core/djangoapps/user_authn/views/login_form.py(216)login_and_registration_form()
-> if request.user.is_authenticated and redirect_to:
(Pdb) l
211  	        breakpoint()
212  	        # This is to handle a case where a logged-in cookie is not present but the user is authenticated.
213  	        # Note: If we don't handle this learner is redirected to authn MFE and then back to dashboard
214  	        # instead of the desired redirect URL (e.g. finish_auth) resulting in learners not enrolling
215  	        # into the courses.
216  ->	        if request.user.is_authenticated and redirect_to:
217  	            return redirect(redirect_to)
218
219  	        query_params = request.GET.urlencode()
220  	        url_path = '/{}{}'.format(
221  	            initial_mode,
(Pdb) saml_provider
False
(Pdb) tpa_hint_provider
(Pdb) enterprise_customer
```